fuzzers/baby/baby_fuzzer_custom_input/src/main.rs

```diff
@@ -132,37 +137,78 @@ pub fn main() {
         .generate_initial_inputs(&mut fuzzer, &mut executor, &mut generator, &mut mgr, 8)
         .expect("Failed to generate the initial corpus");
 
-    let custom_mapped_mutators = havoc_mutations_no_crossover()
-        .merge(havoc_crossover_with_corpus_mapper(
-            &CustomInput::byte_array_mutvec_mapper_corpus_extractor,
-        ))
-        .map(ToWrapsReferenceFunctionMappingMutatorMapper::<
+    #[cfg(feature = "simple_interface")]
+    let (mapped_mutators, optional_mapped_mutators) = {
+        // Creating mutators that will operate on input.byte_array
+        // For now, due to a limitation in lifetime management (see the MappedInput trait),
+        // the types have to be partially specified
+        let mapped_mutators: (MappedInputFunctionMappingMutator<_, _, MutVecInput<'_>>, _) =
```

Comment:
I don't really understand why the extra type specifyer is necessary here?

Reply:
MappedInput maps the type to itself. The compiler can't deal with that.

I would be very happy about a better solution, but I could not find one, even after way too much time spent trying. Feel free to give it a shot though :)

Reply:
Could we get around the extra type if we always reuturn a BytesSubInput?

Reply:
`BytesSubInput` is significantly less general than `WrapsInput`, because it only works for byte array style inputs. It would probably also introduce a performance penalty (because of the extra logic e.g. compared to `MutVecInput`, which is just a plain wrapper).